typescript/packages/planning-server/src/index.ts

```diff
@@ -1,122 +1,260 @@
+        console.log(
+          `${timestamp()} ${requestId} ${colors.green}⚡️ Cache hit!${colors.reset} | ${timeTrack(startTime)}`,
+        );
+        const lastMessage =
+          cachedResult.messages[cachedResult.messages.length - 1];
         return new Response(JSON.stringify(lastMessage), {
           headers: { "Content-Type": "application/json" },
         });
       }
 
-      console.log("Generating:", description);
+      const modelConfig = MODELS[payload.model];
+      if (!modelConfig) {
+        console.warn(
+          `${timestamp()} ${requestId} ${colors.yellow}⚠️  Unsupported model:${colors.reset} ${payload.model}`,
+        );
+      }
+
+      console.log(
+        `${timestamp()} ${requestId} ${colors.blue}🚀 Starting generation${colors.reset} | ${timeTrack(startTime)}`,
+      );
 
       let messages = payload.messages;
 
       let params = {
-        model: payload.model,
-        system: payload.system,
-        maxTokens: payload.max_tokens,
+        model: modelConfig.model || payload.model,
         messages,
-      }
+        stream: payload.stream,
+      } as {
+        model: any;
+        messages: Array<{ role: string; content: string }>;
+        stream: boolean;
+        system?: string;
+        stopSequences?: string[];
+      };
 
-      let modelProvider;
-      if (payload.model.startsWith('groq:')) {
-        modelProvider = groq(payload.model.replace('groq:', ''));
+      // NOTE(jake): Unfortunately the o1 model is a unique snowflake, and requires
+      // a distinctly different request payload than the other models..
+      //
+      // We can't send a system prompt, stop sequences, or max_tokens.
+      if (payload.model?.startsWith("openai:o1")) {
+        if (payload.system && messages.length > 0) {
+          messages[0].content = `${payload.system}\n\n${messages[0].content}`;
+        }
       } else {
-        modelProvider = anthropic(payload.model);
+        params = {
+          ...params,
+          system: payload.system,
+          stopSequences: payload.stop ? [payload.stop] : undefined,
+        };
       }
 
-      const llmStream = await streamText({
-        ...params,
-        model: modelProvider,
-        stopSequences: payload.stop ? [payload.stop] : undefined,
-      });
+      const llmStream = await streamText(params);
 
       let result = "";
-
-      if (messages[messages.length - 1].role === "assistant") {
-        result = messages[messages.length - 1].content;
-      }
+      let tokenCount = 0;
 
       if (payload.stream) {
         const stream = new ReadableStream({
           async start(controller) {
-            // NOTE: the llm doesn't send text we put into its mouth, so we need to
-            // manually send it so that streaming client sees everything assistant 'said'
-            if (messages[messages.length - 1].role === "assistant") {
-              controller.enqueue(new TextEncoder().encode(JSON.stringify(result) + '\n'));
```

Comment:
we still need this code!

this is what returns what you put into the llms mouth